internal/authorization/method_matcher.go

```diff
@@ -0,0 +1,35 @@
+package authorization
+
+import (
+	"github.com/authelia/authelia/internal/configuration/schema"
+	"github.com/authelia/authelia/internal/logging"
+	"github.com/authelia/authelia/internal/utils"
+)
+
+var warningShown = false
+
+// isMethodMatching check whether request method is in the rules methods
+func isMethodMatching(method []byte, methods []string) (matching bool) {
+	log := logging.Logger()
+	// If no method is provided in the rule, we match any method
+	if len(methods) == 0 {
+		log.Trace("method matching skipped as no methods have been defined for the access_control rule")
+		return true
+	}
+
+	if method == nil {
+		if !warningShown {
```

Comment:
Rather than logging a warning, you can check whether X-Forwarded-Method is required by checking if at least one rule needs it. In that case displaying an error for every request is just fine. Otherwise, no message will be displayed.